src/transformation/utils/lualib.ts

```diff
@@ -0,0 +1,28 @@
+import * as ts from "typescript";
+import * as lua from "../../LuaAST";
+import { LuaLibFeature } from "../../LuaLib";
+import { getOrUpdate } from "../../utils";
+import { TransformationContext } from "../context";
+
+const luaLibFeatures = new WeakMap<TransformationContext, Set<LuaLibFeature>>();
+
+export { LuaLibFeature };
+
+export function getUsedLuaLibFeatures(context: TransformationContext): Set<LuaLibFeature> {
+    return getOrUpdate(luaLibFeatures, context, () => new Set());
+}
+
+export function importLuaLibFeature(context: TransformationContext, feature: LuaLibFeature): void {
+    getUsedLuaLibFeatures(context).add(feature);
+}
+
+export function transformLuaLibFunction(
+    context: TransformationContext,
+    feature: LuaLibFeature,
+    tsParent?: ts.Expression,
+    ...params: lua.Expression[]
+): lua.CallExpression {
+    importLuaLibFeature(context, feature);
```

Comment:
Am I missing something here? Didnt we have a mechanism for lualib dependencies?

Reply:
It's done during printing